test/successful-directories.test.js

```diff
@@ -300,15 +303,53 @@ describe('cosmiconfig', () => {
       }
     );
 
+    testSyncAndAsync('finds JS file traversing from cwd', sync => () => {
+      function readFile(searchPath) {
+        switch (searchPath) {
+          case absolutePath('a/b/c/d/e/f/package.json'):
+          case absolutePath('a/b/c/d/e/f/.foorc'):
+          case absolutePath('a/b/c/d/e/f/foo.config.js'):
+          case absolutePath('a/b/c/d/e/package.json'):
+          case absolutePath('a/b/c/d/e/.foorc'):
+            throw { code: 'ENOENT' };
+          case absolutePath('a/b/c/d/e/foo.config.js'):
+            return 'module.exports = { found: true };';
+          default:
+            throw new Error(`irrelevant path ${searchPath}`);
+        }
+      }
+      const readFileMock = mockReadFile(sync, readFile);
+      process.cwd = jest.fn(() => absolutePath('a/b/c/d/e/f'));
```

Comment:
Do we need to revert this change in a `finally`?

Reply:
I'm not super familiar with testing frameworks, but does [`afterEach`](https://github.com/davidtheclark/cosmiconfig/pull/108/files/5401cba51c0a1887b6a32e7012f86eb68e4e1e6f#diff-4b7ed0b193d73a1b1fa66c79c737b673R30) not cut it?

Reply:
My apologies, I did not see that. `afterEach` will suffice. Though I have some reservations about using the test hooks for this since we only mock `process.cwd` in one test.

Reply:
makes sense, i'll wrap it in a try/catch